cmd/system-probe/modules/network_tracer.go

```diff
@@ -10,85 +9,52 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/utils"
 	"github.com/DataDog/datadog-agent/pkg/network"
-	"github.com/DataDog/datadog-agent/pkg/process/statsd"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 
 	"github.com/DataDog/datadog-agent/pkg/ebpf"
 	"github.com/DataDog/datadog-agent/pkg/network/encoding"
 	"github.com/DataDog/datadog-agent/pkg/process/config"
-	"github.com/DataDog/datadog-agent/pkg/process/net"
 )
 
 // ErrSysprobeUnsupported is the unsupported error prefix, for error-class matching from callers
 var ErrSysprobeUnsupported = errors.New("system-probe unsupported")
 
 var inactivityLogDuration = 10 * time.Minute
 
-// SystemProbe maintains and starts the underlying network connection collection process as well as
-// exposes these connections over HTTP (via UDS)
-type SystemProbe struct {
-	cfg *config.AgentConfig
+// NetworkTracer is a factory for NPM's tracer
+var NetworkTracer = api.Factory{
+	Name: "network_tracer",
+	Fn: func(cfg *config.AgentConfig) (api.Module, error) {
+		// Checking whether the current OS + kernel version is supported by the tracer
+		if supported, msg := ebpf.IsTracerSupportedByOS(cfg.ExcludedBPFLinuxVersions); !supported {
```

Comment:
This is general enough to be shared across all eBPF checks, no?

Reply:
Perhaps we can move this logic upstream - it includes checks to ensure that we're not running eBPF on problematic kernels that can run into kernel panics.

Reply:
Yes, I totally agree that this should be unified somehow, but I think it would be better to leave this for another PR. I think the next thing we should tackle is the is cleaning up the `ebpf` package and making sure we have good general-purpose helpers there to ease the development of new eBPF based modules.